clickhouse_search/models.py

```diff
@@ -69,6 +72,249 @@ def __init__(self, name, select='*', order_by=None):
+                    transcript_filters.append({'extendedIntronicSpliceRegionVariant': (1, '{field} = {value}')})
+            elif field in [MOTIF_FEATURES_KEY, REGULATORY_FEATURES_KEY]:
+                filter_qs.append(Q(**{f'sorted_{field}_consequences__array_exists': {
+                    'consequenceTerms': (value, 'hasAny({value}, {field})'),
+                }}))
+            elif field == SPLICE_AI_FIELD:
+                filter_qs.append(cls._get_in_silico_score_q(SPLICE_AI_FIELD, value))
+            elif field == SCREEN_KEY:
+                filter_qs.append(Q(screen_region_type__in=value))
+            elif field not in SV_ANNOTATION_TYPES:
+                allowed_consequences += value
+
+        non_canonical_consequences = [c for c in allowed_consequences if not c.endswith('__canonical')]
+        if non_canonical_consequences:
+            transcript_filters.append(cls._consequence_term_filter(non_canonical_consequences))
+
+        canonical_consequences = [
+            c.replace('__canonical', '') for c in allowed_consequences if c.endswith('__canonical')
+        ]
+        if canonical_consequences:
+            transcript_filters.append(
+                cls._consequence_term_filter(canonical_consequences, canonical=(0, '{field} > {value}')),
+            )
+
+        return filter_qs, transcript_filters
+
+    @staticmethod
+    def _consequence_term_filter(consequences, **kwargs):
+        return {'consequenceTerms': (consequences, 'hasAny({value}, {field})'), **kwargs}
+
+    @staticmethod
+    def _hgmd_filter_q(hgmd):
+        min_class = next((class_name for value, class_name in HGMD_CLASS_FILTERS if value in hgmd), None)
+        max_class = next((class_name for value, class_name in reversed(HGMD_CLASS_FILTERS) if value in hgmd), None)
+        if 'hgmd_other' in hgmd:
+            min_class = min_class or 'DP'
+            max_class = None
+        if min_class == max_class:
+            return Q(hgmd__classification=min_class)
+        elif min_class and max_class:
+            return Q(hgmd__classification__range=(min_class, max_class))
+        return Q(hgmd__classification__gt=min_class)
+
+    @classmethod
+    def _clinvar_filter_q(cls, clinvar_filters, _get_range_q=None):
+        ranges = [[None, None]]
+        for path_filter, start, end in CLINVAR_PATH_RANGES:
+            if path_filter in clinvar_filters:
+                ranges[-1][1] = end
+                if ranges[-1][0] is None:
+                    ranges[-1][0] = start
+            elif ranges[-1] != [None, None]:
+                ranges.append([None, None])
+        ranges = [r for r in ranges if r[0] is not None]
+
+        clinvar_qs = [(_get_range_q or cls._clinvar_range_q)(path_range) for path_range in ranges]
+        clinvar_q = clinvar_qs[0]
+        for q in clinvar_qs[1:]:
+            clinvar_q |= q
+        return clinvar_q
+
+    @classmethod
+    def _clinvar_range_q(cls, path_range):
```

Comment:
this changed as we are now querying from the clinvar column that was joined on the entry subquery and not the clinvar table itself